### PR TITLE
Feature/datetime subtypes

### PR DESCRIPTION
**Pull request by rosastern**

### CODE REVIEW

File: src/entity/builtin_entity.rs [hunk 1]
```diff
@@ -671,6 +807,27 @@ impl BuiltinEntityKind {
...
+                        BuiltinEntityKind::Time |
+                        BuiltinEntityKind::DatePeriod |
+                        BuiltinEntityKind::TimePeriod => BuiltinEntityKind::Datetime,
+                        _ => builtin_entity_kind,
+                    }
+                }
+            };
+        eprintln!("map_to_supported({:?}, {:?}) => {:?}", builtin_entity_kind, language, result);
+        result
+    }
```
Comment (adrienball) [inline src/entity/builtin_entity.rs]: I'm not sure why we need to remap here. The notion of supporting a builtin entity in a given language is relative to parsing capabilities (is there a parser in english for temperatures?), so it shouldn't affect types and mapping.
Said differently, if a builtin entity is not supported in a certain language, then it won't be output, so no need to remap it to something else.

File: src/entity/builtin_entity.rs [hunk 2]
```diff
@@ -656,6 +780,18 @@ impl BuiltinEntityKind {
...
+            ],
+            BuiltinEntityKind::Time => &[
+                Language::EN,
+            ],
+            BuiltinEntityKind::DatePeriod => &[
+                Language::EN,
+            ],
+            BuiltinEntityKind::TimePeriod => &[
+                Language::EN,
+            ],
```
Comment (adrienball) [inline src/entity/builtin_entity.rs]: Code style detail: you can put short lists on a single line for compactness.

File: src/entity/builtin_entity.rs [hunk 3]
```diff
@@ -114,11 +122,15 @@ impl BuiltinEntityKind {
     pub fn description(&self) -> &'static str {
         match *self {
             BuiltinEntityKind::AmountOfMoney => "Matches an amount of money",
-            BuiltinEntityKind::Duration => "Matches a time duration",
+            BuiltinEntityKind::Duration => "Matches a date/time duration",
```
[the remark this answers is not in the document]
Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅

File: src/entity/builtin_entity.rs [hunk 4]
```diff
@@ -562,6 +660,32 @@ impl BuiltinEntityKind {
                     to: Some("2017-06-08 00:00:00 +02:00".to_string()),
                 }),
             ]),
+            BuiltinEntityKind::Date => serde_json::to_string_pretty(&vec![
+                SlotValue::InstantTime(InstantTimeValue {
+                    value: "2017-06-13 18:00:00 +02:00".to_string(),
+                    grain: Grain::Hour,
+                    precision: Precision::Exact,
+                }),
+            ]),
```
[the remark this answers is not in the document]
Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅

File: src/entity/builtin_entity.rs [hunk 5]
```diff
@@ -562,6 +660,32 @@ impl BuiltinEntityKind {
...
+            BuiltinEntityKind::DatePeriod => serde_json::to_string_pretty(&vec![
+                SlotValue::TimeInterval(TimeIntervalValue {
+                    from: Some("2017-06-07 18:00:00 +02:00".to_string()),
+                    to: Some("2017-06-09 18:00:00 +02:00".to_string()),
+                }),
```
[the remark this answers is not in the document]
Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅

File: src/entity/builtin_entity.rs (hunk 1 above)
Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅ 
Rustling-ontology tagging modified to have the support work like that.

File: src/entity/builtin_entity.rs (hunk 2 above)
Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅

File: src/entity/builtin_entity.rs [hunk 6]
```diff
@@ -114,11 +122,15 @@ impl BuiltinEntityKind {
     pub fn description(&self) -> &'static str {
         match *self {
             BuiltinEntityKind::AmountOfMoney => "Matches an amount of money",
-            BuiltinEntityKind::Duration => "Matches a time duration",
+            BuiltinEntityKind::Duration => "Matches a  time duration",
```
Comment (adrienball) [inline src/entity/builtin_entity.rs]: extra space here

Comment (rosastern) [reply src/entity/builtin_entity.rs]: ✅

**Pull request closed without merging**